tls.go

```diff
@@ -39,23 +33,30 @@ func initTLS(cfg *Config) (*tls.Config, error) {
 		}
 
 		return &tls.Config{
-			MinVersion:   tls.VersionTLS12,
-			ClientAuth:   cfg.TLS.auth,
-			Certificates: []tls.Certificate{cert},
-			ClientCAs:    certPool,
-			RootCAs:      certPool,
-			ServerName:   cfg.TLS.ServerName,
+			MinVersion:           tls.VersionTLS12,
+			ClientAuth:           cfg.TLS.auth,
+			GetClientCertificate: getClientCertificate(cfg),
+			ClientCAs:            certPool,
+			RootCAs:              certPool,
+			ServerName:           cfg.TLS.ServerName,
 		}, nil
 	}
 
-	cert, err = tls.LoadX509KeyPair(cfg.TLS.Cert, cfg.TLS.Key)
-	if err != nil {
-		return nil, err
-	}
-
 	return &tls.Config{
-		ServerName:   cfg.TLS.ServerName,
-		MinVersion:   tls.VersionTLS12,
-		Certificates: []tls.Certificate{cert},
+		ServerName:           cfg.TLS.ServerName,
+		MinVersion:           tls.VersionTLS12,
+		GetClientCertificate: getClientCertificate(cfg),
 	}, nil
 }
+
+// getClientCertificate is used for tls.Config struct field GetClientCertificate and enables re-fetching the client certificates when they expire
+func getClientCertificate(cfg *Config) func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
`info` is unused, should be _, or removed

Reply:
I meant, both `infos` 😃 They're both unused.

Reply:
sorry! Missed that one 🤦 fixed